### ISSUE TITLE
Data driven style not working in built-in Mapbox GL viewer

### ISSUE THREAD

**Issue by kiambogo** [pka/t-rex#25]
I need to access a row returned from the executed SQL statement in a layer, and use it in the styling for that layer. 

For example, lets say I have a layer defined as the following:
```
[[tileset]]
name = "osm"

[[tileset.layer]]
name = "line"
geometry_field = "geometry"
geometry_type = "LINESTRING"
fid_field = "osm_id"
simplify = true
  [[tileset.layer.query]]
  sql = """
    SELECT name, type, id as osm_id, ST_Union(geometry) AS geometry, colour
    FROM my_geom_table"""
```

where `colour` here is the hex value as text of the colour for that line (i.e. `#ff00ff`).

Then, in my styling guide for the layer, I need to set the colour for each line to be its value as is returned from the db. Ideally, something like this:

```
[[tileset.layer]]
name = "line"
# ...
[tileset.layer.style]
  type = "line"
  [tileset.layer.style.paint]
  line-color = [colour]
```

I have this functionality in Mapnik (moving from PNG raster tiles to vector tiles), as the following:
https://github.com/trifectalabs/roadquality/blob/master/resources/tilestache/palette.mss#L98
in my style guide, and `rating_colour` is referencing the returned row from the executed SQL: https://github.com/trifectalabs/roadquality/blob/master/resources/tilestache/project.mml#L240

I realize that this may not be as t-rex specific, but maybe more the TOML implementation of the Mapbox Style Spec, but can you offer any insight whether this is feasible, or something that could be added to the spec?

Thanks!

**Comment by pka** [comment]
I think, Mapbox GL doesn't offer this functionality You can set a color as function of a field value, but I don't find a way to use the field value as color directly.
```
[tileset.layer.style]
  type = "line"
  [tileset.layer.style.paint.line-color]
  property = "z_order"
  stops = [{ in = 0, out = "#d8e8c8" }, { in = 3, out = "#cc0000" }]
```

BTW, when you click on the Maputnik link in the backend, you can experiment with the Mapbox style and translate the generated JSON to TOML afterwards.

**Comment by pka** [comment]
Correction: according to https://www.mapbox.com/mapbox-gl-js/example/data-driven-lines/ it could be possible:
```
[tileset.layer.style]
  type = "line"
  [tileset.layer.style.paint.line-color]
  property = "colour"
  type = "identity"
```
But when testing with a string field, I always get `TypeError: m is undefined`

**Comment by kiambogo** [comment]
After hitting my for a while, I found out the issue is just in the Mapbox GL viewer within your `viewer.js`. I updated the viewer to use Mapbox GL JS 0.38, but that still did not fix it. If I use vanilla mapbox 0.38 in a static HTML file pointing to t-rex (like the snippet), I get the expected result. 

Huge thanks for your help! :D

**Comment by pka** [comment]
Will try to fix that when updating the built-in Mapbox viewer.

**Comment by pka** [comment]
Fixed in 9641c7a
